hibernate-core/src/test/java/org/hibernate/test/hql/size/WhereClauseOrderBySizeTest.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+
+package org.hibernate.test.hql.size;
+
+import org.hibernate.annotations.ResultCheckStyle;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+import org.hibernate.dialect.DB2Dialect;
+import org.hibernate.dialect.Oracle12cDialect;
+import org.hibernate.dialect.SQLServerDialect;
+import org.hibernate.jpa.test.BaseEntityManagerFunctionalTestCase;
+
+import org.hibernate.testing.SkipForDialect;
+import org.hibernate.testing.SkipForDialects;
+import org.hibernate.testing.TestForIssue;
+import org.junit.Test;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.TypedQuery;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.hibernate.testing.transaction.TransactionUtil.doInJPA;
+import static org.junit.Assert.assertEquals;
+
+@TestForIssue(jiraKey = "HHH-14585")
+@SkipForDialects({
+		@SkipForDialect(value = SQLServerDialect.class, comment = "The @SQLDelete expression seems to be not working with this dialect"),
+		@SkipForDialect(value = DB2Dialect.class, comment = "The @SQLDelete expression seems to be not working with this dialect"),
+		@SkipForDialect(value = Oracle12cDialect.class, comment = "The @SQLDelete expression seems to be not working with this dialect")
```

Comment:
Please update the comments to say something like `The database does not support boolean data types`. I think `WhereAnnotatedOneToManySizeTest` works on SQL Server and DB2 because `true`/`false` is an alias for `1`/`0`. For updating, this might not work though as you seem to have discovered.

Reply:
Thanks @beikov , Fabio and I were about to ask about the db2 issue :)